packages/expo-task-manager/src/TaskManager.ts

```diff
@@ -86,13 +86,15 @@ export interface RegisteredTask extends TaskManagerTask {}
 /**
  * Type of task executor – a function that handles the task.
  */
-export type TaskManagerTaskExecutor<T = unknown> = (body: TaskManagerTaskBody<T>) => void;
+export type TaskManagerTaskExecutor<T = any> = (body: TaskManagerTaskBody<T>) => Promise<any>;
```

Comment:
rather than Promise<any> it may be better to have 

```
export type TaskManagerTaskExecutor<T = unknown, R= unknown> = (body: TaskManagerTaskBody<T>) => Promise<R>;
```

Reply:
I was thinking about introducing an another generic, but if you look at the code, we already were casting the result to `any`, and it would be a breaking change for all of the task manager users in the TS project, and since we are close to the stable SDK release I decided not to introduce breaking change that late in the process.

However, we might revisit it this in the future, and follow up with more strict typing.

Reply:
I'm talking about the typing which we apply for the returned result in the handling process, it was `any` for few years already, and I have changed `unknown` to match that:
* https://github.com/expo/expo/blame/6da22921f35db183fc6722c1b697ff77d820387c/packages/expo-task-manager/src/TaskManager.ts#L259

Reply:
Ok ...

I was wondering why `T=unknown` changed to `T=any`?

Also 

adding a typing after so it's `<T=unknown,R=unknown>` would not break the type because `Promise<R>` which resolves to `R` would get matched with `any`

What it does allow you to do is for `BackgroundFetch` as an example in a future API

`BackgroundFetch.defineTask(task:TaskManagerExecutor<Record<string,unknown>, BackgroundFetchResult>)`

So it's a specific strongly typed version of defineTask for BackgroundFetch.

I chose Record<string,unknown> because the data varies from APNS vs FCM as I have noted in

- https://github.com/trajano/expo-experiments/blob/master/packages/my-app/src/notifications/extractDataFromBackgroundNotification.ts
- https://github.com/trajano/expo-experiments/blob/master/packages/my-app/src/notifications/extractDataFromBackgroundNotification.android.ts